server/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -202,9 +203,27 @@ private synchronized boolean assertNoSeqNumberConflict(long seqNo, BytesReferenc
             if (previous.v1().equals(data) == false) {
                 Translog.Operation newOp = Translog.readOperation(new BufferedChecksumStreamInput(data.streamInput()));
                 Translog.Operation prvOp = Translog.readOperation(new BufferedChecksumStreamInput(previous.v1().streamInput()));
-                throw new AssertionError(
-                    "seqNo [" + seqNo + "] was processed twice in generation [" + generation + "], with different data. " +
-                        "prvOp [" + prvOp + "], newOp [" + newOp + "]", previous.v2());
+                // we need to exclude versionType from this check because it's removed in 7.0
+                final boolean sameOp;
+                if (prvOp instanceof Translog.Index && newOp instanceof Translog.Index) {
```

Comment:
versionType does not exist anymore in `Translog.Index` and `Translog.Delete`, so can't we just use `equals` on the `Translog.Operation` that we recreated from the bytes?